tests/polynomial/elementwise_to_affine.mlir

```diff
@@ -1,4 +1,7 @@
 // RUN: heir-opt --convert-elementwise-to-affine %s | FileCheck --enable-var-scope %s
+// RUN: heir-opt '--convert-elementwise-to-affine=restrict-ops=polynomial.add,polynomial.mul_scalar  restrict-dialects=polynomial' %s | FileCheck --enable-var-scope %s
+// RUN: heir-opt --convert-elementwise-to-affine=restrict-dialects=polynomial %s | FileCheck --enable-var-scope %s
+// RUN: heir-opt --convert-elementwise-to-affine=restrict-ops=polynomial.add,polynomial.mul_scalar %s | FileCheck --enable-var-scope %s
 
```

Comment:
I've added some tests in the file. please let me know what you think.

Reply:
The test RUNs themselves look good to me now! If I can just suggest testing the positive conditions with the CHECK_POLY_MUL/CHECK_POLY_ADD - I think right now you are checking the cases of poly mul/add *not* being converted, but it might be better to check that the ops being restricted to are also being selectively converted.

In that way though you wouldn't be able to combine both CHECK-* conditions on the both case, which is maybe why you had done it that way.

Reply:
Here's a version that doesn't require the separate test functions at the bottom:

https://gist.github.com/AlexanderViand-Intel/2cee3b1b816028a83c4516935974d63a

Reply:
Thanks for investing so much time. There's a lot to learn about FileCheck !